core/src/main/java/com/graphhopper/routing/util/WheelchairFlagEncoder.java

```diff
@@ -0,0 +1,209 @@
+        avoidHighwayTags.add("secondary_link");
+        avoidHighwayTags.add("tertiary");
+        avoidHighwayTags.add("tertiary_link");
+        avoidHighwayTags.add("steps");
+        avoidHighwayTags.add("track");
+
+        avoidSurfaces.add("cobblestone");
+        avoidSurfaces.add("gravel");
+        avoidSurfaces.add("sand");
+
+        avoidSmoothness.add("bad");
+        avoidSmoothness.add("very_bad");
+        avoidSmoothness.add("horrible");
+        avoidSmoothness.add("very_horrible");
+        avoidSmoothness.add("impassable");
+
+        // for now no explicit avoiding #257
+        //avoidHighwayTags.add("cycleway"); 
+        allowedHighwayTags.addAll(safeHighwayTags);
+        allowedHighwayTags.addAll(avoidHighwayTags);
+        allowedHighwayTags.add("cycleway");
+        allowedHighwayTags.add("unclassified");
+        allowedHighwayTags.add("road");
+        // disallowed in some countries
+        //allowedHighwayTags.add("bridleway");
+
+        maxPossibleSpeed = FERRY_SPEED;
+        speedDefault = MEAN_SPEED;
+        init();
+    }
+
+    @Override
+    public int getVersion() {
+        return 1;
+    }
+
+    /**
+     * Avoid some more ways than for pedestrian like hiking trails.
+     */
+    @Override
+    public EncodingManager.Access getAccess(ReaderWay way) {
+        if (way.hasTag("wheelchair", intendedValues)) {
+            return EncodingManager.Access.WAY;
+        }
+
+        if (way.getTag("sac_scale") != null) {
+            return EncodingManager.Access.CAN_SKIP;
+        }
+
+        if (way.hasTag("highway", avoidHighwayTags) && !way.hasTag("sidewalk", sidewalkValues)) {
+            return EncodingManager.Access.CAN_SKIP;
+        }
+
+        if (way.hasTag("surface", avoidSurfaces)) {
+            if (!way.hasTag("sidewalk", sidewalkValues)) {
+                return EncodingManager.Access.CAN_SKIP;
+            } else {
+                String sidewalk = way.getTag("sidewalk");
+                if (way.hasTag("sidewalk:" + sidewalk + ":surface", avoidSurfaces)) {
+                    return EncodingManager.Access.CAN_SKIP;
+                }
+            }
+        }
+
+        if (way.hasTag("smoothness", avoidSmoothness)) {
+            if (!way.hasTag("sidewalk", sidewalkValues)) {
+                return EncodingManager.Access.CAN_SKIP;
+            } else {
+                String sidewalk = way.getTag("sidewalk");
+                if (way.hasTag("sidewalk:" + sidewalk + ":smoothness", avoidSmoothness)) {
+                    return EncodingManager.Access.CAN_SKIP;
+                }
+            }
+        }
+
+        return super.getAccess(way);
+    }
+
+    @Override
+    public IntsRef handleWayTags(IntsRef edgeFlags, ReaderWay way, EncodingManager.Access access, long relationFlags) {
+        if (access.canSkip()) {
+            return edgeFlags;
+        }
+
+        if (!access.isFerry()) {
+            String sacScale = way.getTag("sac_scale");
```

Comment:
Why is this required? The way should have `access.canSkip` (see getAccess method)